### ISSUE TITLE
Local Port Mapping not working on Docker Desktop for Linux / Ubuntu

### ISSUE THREAD

**Issue by flash-me** [kubernetes-sigs/kind#2948]
**What happened**:

I deployed a cluster via `kind create cluster` and exposed a service with `NodePort`. 
[According to the docs](https://kind.sigs.k8s.io/docs/user/configuration/#extra-port-mappings), I should be able to access the service through the nodes IP:
![image](https://user-images.githubusercontent.com/5240450/193430228-983d4350-6183-49ab-8393-a0e889513d15.png)

But I am not. 

Here are some details:
![image](https://user-images.githubusercontent.com/5240450/193430289-4a688fa5-c52a-4044-bc9e-9437af5a080a.png)

**What you expected to happen**:

Being able to access the exposed service through the nodes IP

**How to reproduce it (as minimally and precisely as possible)**:

- Deploy cluster
- Deploy kubernetes dashboard
- Change service type from ClusterIP to NodePort
- Try accessing the service with `<nodeIP>:<port>`
- Cry

**Anything else we need to know?**:

Service is accessible and runs and curl'ing from the node itself
![image](https://user-images.githubusercontent.com/5240450/193430322-1aa1d777-8883-4634-8b2e-e48cf890be3f.png)

**Environment:**

- kind version: (use `kind version`): **v0.16.0 go1.19.1 linux/amd64**
- Kubernetes version: (use `kubectl version`): **v1.25.2**
- Docker version: (use `docker info`): Engine: **20.10.18** Docker Desktop: **4.12.0**
- OS (e.g. from `/etc/os-release`): **Ubuntu 22.04.1 LTS**

cheers
flash ⚡

**Comment by BenTheElder** [comment]
Can you access ports on other containers without port forwarding?

docker is responsible for this layer of the networking but in your host environment things like VPN or firewall can interfere.

**Comment by aojea** [comment]
>  Docker Desktop: 20.10.17

??

**Comment by flash-me** [comment]
@aojea you're right, it's 4.12.0, I updated the description. Thanks!

@BenTheElder 
It's my private dev setup, no proxies, no VPN, no firewall. Default Ubuntu installation.

I deployed nginx, same results:
![image](https://user-images.githubusercontent.com/5240450/193447382-86b73f81-206e-4345-a0b1-f7944d29742f.png)

Let me know if you need any kind of information (whether its network interfaces or other stuff)

cheers
flash ⚡

**Comment by flash-me** [comment]
These docker settings are what has been setup from `kind create cluster` 

**docker inspect kind-control-plane**
```json
{
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "40591cf90854e8b758b14bc2cdc787d147e59b5b8deee26805cac2f4cdaccf8e",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {
                "6443/tcp": [
                    {
                        "HostIp": "127.0.0.1",
                        "HostPort": "41123"
                    }
                ]
            },
            "SandboxKey": "/var/run/docker/netns/40591cf90854",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "",
            "Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "MacAddress": "",
            "Networks": {
                "kind": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": [
                        "58224b502158",
                        "kind-control-plane"
                    ],
                    "NetworkID": "07bcde1b0350e13c424d835c2ba7016d65f9ef4fa8bc619f7e123393caf0d5bb",
                    "EndpointID": "045e46a6da5d36b24f5b0c11900305e7ab9d9a54b9e75d8f301e280bc4ddcbdc",
                    "Gateway": "172.18.0.1",
                    "IPAddress": "172.18.0.2",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "fc00:f853:ccd:e793::1",
                    "GlobalIPv6Address": "fc00:f853:ccd:e793::2",
                    "GlobalIPv6PrefixLen": 64,
                    "MacAddress": "02:42:ac:12:00:02",
                    "DriverOpts": null
                }
            }
        }
    }
```


and **docker inspect kind** (network)
```json
{
    "Name": "kind",
    "Id": "07bcde1b0350e13c424d835c2ba7016d65f9ef4fa8bc619f7e123393caf0d5bb",
    "Created": "2022-10-01T20:51:41.732129119Z",
    "Scope": "local",
    "Driver": "bridge",
    "EnableIPv6": true,
    "IPAM": {
        "Driver": "default",
        "Options": {},
        "Config": [
            {
                "Subnet": "172.18.0.0/16",
                "Gateway": "172.18.0.1"
            },
            {
                "Subnet": "fc00:f853:ccd:e793::/64",
                "Gateway": "fc00:f853:ccd:e793::1"
            }
        ]
    },
    "Internal": false,
    "Attachable": false,
    "Ingress": false,
    "ConfigFrom": {
        "Network": ""
    },
    "ConfigOnly": false,
    "Containers": {
        "58224b502158037b5182a2b5f381cc31893b62f5b892cc23c6da24ece5aab7e2": {
            "Name": "kind-control-plane",
            "EndpointID": "045e46a6da5d36b24f5b0c11900305e7ab9d9a54b9e75d8f301e280bc4ddcbdc",
            "MacAddress": "02:42:ac:12:00:02",
            "IPv4Address": "172.18.0.2/16",
            "IPv6Address": "fc00:f853:ccd:e793::2/64"
        }
    },
    "Options": {
        "com.docker.network.bridge.enable_ip_masquerade": "true",
        "com.docker.network.driver.mtu": "1500"
    },
    "Labels": {}
}
```


So my question is how the statement from the documentation page can even work? 
By default it is a bridge network and every port I want to access in my docker network must be exposed explicitly, or am I missing something?

cheers
flash ⚡

**Comment by BenTheElder** [comment]
The fact that you can't access the standalone Nguni container through its IP on Linux says something is wrong with your docker environment.

**Comment by aojea** [comment]
but I can not understand how you can have docker desktop and linux, "Docker Desktop is an application for MacOS and Windows "

**Comment by flash-me** [comment]
> but I can not understand how you can have docker desktop and linux, "Docker Desktop is an application for MacOS and Windows "

https://docs.docker.com/desktop/install/linux-install/

cheers
flash ⚡

**Comment by aojea** [comment]
Today I Learned 

It seems it installs a VM, especulating, all network limitations for Mac and Windows apply, since there is no direct connectivity from the host to the containers inside the VM

**Comment by flash-me** [comment]
> Today I Learned
> 
> It seems it installs a VM, speculating, all network limitations for Mac and Windows apply, since there is no direct connectivity from the host to the containers inside the VM
> 
> https://docs.docker.com/desktop/install/linux-install/

Yes also figured that out right now. I can ensure the same limitations.
After I removed docker desktop and did a clean reinstall of docker-ce everything works es expected!
![image](https://user-images.githubusercontent.com/5240450/193469596-7b56580b-b6d4-47f2-8a4f-2f960ca2d0b7.png)


You might want to update the docs that the statement is only true for non docker-desktop installs.

Thanks for any help @BenTheElder and @aojea 

cheers
flash ⚡

**Comment by flash-me** [comment]
The same for my initial issue with accessing the node of kind. Works
![image](https://user-images.githubusercontent.com/5240450/193469811-0defe0da-5b26-4b3d-af8a-96119e60383b.png)

From my side this issue can be closed, except you want to track it for the docker desktop limitations.

cheers
flash ⚡

**Comment by BenTheElder** [comment]
Thanks for digging into this.

That's ... surprising. FWIW this is going to have significant tradeoffs on linux and for most docker usage I think you'll find it easier to just install docker from the apt package. A huge benefit to working with containers on linux is direct access to everything including the network.

**Comment by BenTheElder** [comment]
Clearly will need to clarify the docs on our end to address all docker desktop app installs on all platforms.

**Comment by flash-me** [comment]
> Thanks for digging into this.
> 
> That's ... surprising. FWIW this is going to have significant tradeoffs on linux and for most docker usage I think you'll find it easier to just install docker from the apt package. A huge benefit to working with containers on linux is direct access to everything including the network.

Yes. TBH I did not expect that the Docker Desktop also installs a VM even on Linux (why though?). I'm totally fine with docker-engine only, but yes, must be clear for the users, not only in case of _kind_.

cheers
flash ⚡

**Comment by flash-me** [comment]
Can be closed afer #2950

**Comment by BenTheElder** [comment]
thank you!
